src/content-scripts/attention-collector.js

```diff
@@ -139,25 +142,121 @@
          */
         let ogType = '';
 
-        let attentionStartTime;
-        let attentionStopTime;
-        let audioStartTime;
-        let audioStopTime;
-        let scrollHeight = 0;
-        let maxPixelScrollDepth = 0;
+                /**
+         * The canonical url or og:url tags in the page's head.
+         * @type {string}
+         */
+        let canonicalOrOGURL = '';
+
+        /** 
+         * The start time (unix timestamp in ms) the attention event started.
+         * @type {number}
+         */
+        let attentionStartTime = 0;
+        /** 
+         * The start time (unix timestamp in ms) the attention event stopped.
+         * @type {number}
+         */
+        let attentionStopTime = 0;
+        /** 
+         * The start time (unix timestamp in ms) the audio event started.
+         * @type {number}
+         */
+        let audioStartTime = 0;
+        /** 
+         * The start time (unix timestamp in ms) the audio event stopped.
+         * @type {number}
+         */
+        let audioStopTime = 0;
+
+        // DOM ELEMENT COLLECTION FUNCTIONS
+        /**
+         * 
+         * @param {*} documentElement 
+         * @returns {string} the content or innerText of the selected DOM element
+         */
+        function getContentsHavingSelector(str, documentElement) {
+            const e = documentElement.querySelector(str);
+            return e === null ? undefined : e.content || e.innerText;
+        }
+
+        /**
+         * 
+         * @param {*} documentElement 
+         * @returns {string} the content of the title element
+         */
+        function getTitle(documentElement) {
+            return getContentsHavingSelector('title', documentElement);
+        }
+        
+        /**
+         * 
+         * @param {*} documentElement 
+         * @returns {string} the content of the meta og:type tag
+         */
+        function getOGType(documentElement) {
+            return getContentsHavingSelector('meta[property="og:type"]', documentElement);
+        }
 
+        function getCanonicalURL(documentElement) {
+            const elem = documentElement.querySelector('link[rel="canonical"]');
+            if (elem === null) return undefined;
+            return elem.href;
```

Comment:
Consider using a ternary here, that'll make it even more concise than the braceless-`if` :)
`return elem === null ? undefined : elem.href`

Reply:
Otherwise please put braces around the `if` I misread this at first and thought it was just a formatting problem :)